pgxn/neon/control_plane_connector.c

```diff
@@ -161,7 +162,10 @@ ConstructDeltaMessage()
 			PushKeyValue(&state, "name", entry->name);
 			if (entry->password)
 			{
+				char	   *logdetail;
+
 				PushKeyValue(&state, "password", (char *) entry->password);
+				PushKeyValue(&state, "encrypted_password", get_role_password(entry->name, &logdetail));
```

Comment:
From a description `get_role_password` may return an error. Makes sense to log it to help with debugging things. Also `PushKeyValue` calls `strlen` on the result which is null on error, IIRC that will result in crash

Reply:
Should it just log or fail the query?

Reply:
Yes, I think it is better to error out since we can reasonably proceed without it. Function comment says `The error reason should *not* be sent to the client, to avoid giving away user information!` but I think that is mostly about unauthenticated users and in our case it is ok to show the reason.

Reply:
Sounds good, that was my thinking as well. I  made it error out and print the message. `elog` does a `longjmp` right? So I don't have to return early?

Reply:
> Sounds good, that was my thinking as well. I made it error out and print the message. elog does a longjmp right? So I don't have to return early?

yes, `elog(ERROR, ...)` should about whole transaction.